zerver/migrations/0574_backfill_directmessagegroup_group_size.py

```diff
@@ -0,0 +1,72 @@
+# Generated by Django 5.0.6 on 2024-08-16 06:44
+
+from django.db import migrations, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Count, Model, OuterRef, Subquery
+
+BATCH_SIZE = 1000
+
+
+def backfill_group_size_field_for_direct_message_groups(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    direct_message_group_model = apps.get_model("zerver", "DirectMessageGroup")
+    recipient_model = apps.get_model("zerver", "Recipient")
+    subscription_model = apps.get_model("zerver", "Subscription")
+
+    RECIPIENT_DIRECT_MESSAGE_GROUP = 3
+
+    direct_message_group_recipient_entries = recipient_model.objects.filter(
+        type=RECIPIENT_DIRECT_MESSAGE_GROUP
+    )
+
+    if not direct_message_group_recipient_entries.exists():
+        return
+
+    recipient_id_lower_bound = direct_message_group_recipient_entries.earliest("id").id
+    recipient_id_upper_bound = direct_message_group_recipient_entries.latest("id").id
+
+    while recipient_id_lower_bound <= recipient_id_upper_bound:
```

Comment:
This migration will not work correctly, because you're computing the highest ID outside a transaction, and so new direct message groups could be created while the migration is running. We don't want to run the entire migration inside a transaction, so instead we need to make sure we run significantly past the highest ID in the `Recipient` table (not the highest ID that's a direct message group, because new IDs are assigned consecutively without regard to recipient type).

See for example zerver/migrations/0568_mark_narrow_to_dm_with_welcome_bot_new_user_as_read.py.

Reply:
oh, I haven't thought about that.. I'll update it